src/components/settings/PasswordEntry.tsx

```diff
@@ -0,0 +1,100 @@
+import { Button, PasswordInput, Stack, Text, Title } from "@mantine/core";
+import React, { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { Edit, Lock } from "tabler-icons-react";
+
+type Props = {
+  createNew?: boolean;
+  onValidate: (string, boolean) => void;
+  closing?: boolean;
+};
+export function PasswordEntry(props: Props): JSX.Element {
+  const { closing, createNew, onValidate } = props;
+
+  const [editPasswordMode, setEditPasswordMode] = useState(false);
+  const { t } = useTranslation();
+  const [newPassword, setNewPassword] = useState("");
+  const [newPasswordConfirm, setNewPasswordConfirm] = useState("");
+  const [newPasswordError, setNewPasswordError] = useState("");
+  const [confirmPasswordError, setConfirmPasswordError] = useState("");
+
+  useEffect(() => {
+    validateAndUpdatePassword(newPassword, newPasswordConfirm, closing);
+  }, [createNew, closing]);
+
+  const validateAndUpdatePassword = (password, passwordConfirm, closing = false) => {
+    setConfirmPasswordError("");
+    setNewPasswordError("");
+    var validPassword = "";
+    var isValid = false;
+
+    if (password || passwordConfirm) {
+      if (password == passwordConfirm) {
+        validPassword = password;
+        isValid = true;
+        setNewPasswordError("");
+        setConfirmPasswordError("");
+      } else if (passwordConfirm !== "") {
+        setConfirmPasswordError(t("settings.password.errormustmatch"));
+      } else if (closing) {
+        setConfirmPasswordError(t("settings.password.errormustretype"));
+      }
+    } else if (editPasswordMode || createNew) {
+      setNewPasswordError(t("settings.password.errorcannotbeblank"));
+    } else {
+      isValid = true;
+    }
+
+    onValidate(validPassword, isValid);
+  };
+
+  return (
+    <Stack spacing="xs">
+      <Title order={6} style={{ paddingTop: "15px" }}>
+        {createNew ? (
+          <Text>{t("settings.password.titlesetpassword")}</Text>
+        ) : (
+          <Text>
+            {t("settings.password.titlechangepassword")}
+            <Button
```

Comment:
Buttons without Text and just an icon look better as an ActionIcon: https://mantine.dev/core/action-icon/

Reply:
I think you used it also in other places like Admin Area. Would be awesome if you can change it :)

Reply:
Good catch - I didn't realize this was a thing... I've converted them to ActionIcons.